test/Test/Hspec/ExpectationsSpec.hs

```diff
@@ -112,3 +112,8 @@ spec = do
 
     it "fails, if an exception of a specific type is required, but no exception is thrown" $ do
       (return () `shouldThrow` anyErrorCall) `shouldThrow` expectationFailed (Reason "did not get expected exception: ErrorCall")
+
+  describe "orScream" $ do
+    it "prepends a message" $ do
+      ("foo" `shouldBe` "bar" `orScream` "Oh no!")
+        `shouldThrow` expectationFailed (Reason "Oh no!\n\nexpected: \"bar\"\n but got: \"foo\"")
```

Comment:
This test is currently failing with:

```
*** Exception: HUnitFailure (Just (SrcLoc {srcLocPackage = "main", srcLocModule = "Test.Hspec.ExpectationsSpec", srcLocFile = "test/Test/Hspec/ExpectationsSpec.hs", srcLocStartLine = 118, srcLocStartCol = 7, srcLocEndLine = 119, srcLocEndCol = 98})) (Reason "predicate failed on expected exception: HUnitFailure (HUnitFailure (Just (SrcLoc {srcLocPackage = \"main\", srcLocModule = \"Test.Hspec.ExpectationsSpec\", srcLocFile = \"test/Test/Hspec/ExpectationsSpec.hs\", srcLocStartLine = 118, srcLocStartCol = 8, srcLocEndLine = 118, srcLocEndCol = 30})) (Reason \"Oh no!\\n\\nexpected: \\\"bar\\\"\\n but got: \\\"foo\\\"\"))")
```

It seems right to me, but maybe I'm missing some amount of escaping.